src/Agent.Listener/DistributedTask.Pipelines/Yaml/Contracts/CheckoutStep.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.TeamFoundation.DistributedTask.Orchestration.Server.Pipelines.Yaml.TypeConverters;
+
+namespace Microsoft.TeamFoundation.DistributedTask.Orchestration.Server.Pipelines.Yaml.Contracts
+{
+    internal sealed class CheckoutStep : ISimpleStep
+    {
+        public String Name { get; set; }
+
+        internal String Clean { get; set; }
```

Comment:
@nschonni one thing to note with this code.  The goal in the agent is to support a local run scenario for the innerloop development of a given build definition and will have some inherent limitations that we will document.  When you actually run a CI based on a YAML definition it will not appear any different to the agent than a definition that was created using the web editor.